### PR TITLE
switch over to use rouge and GFM

### PR DESCRIPTION
**Pull request by tfoote**
Switch to rouge from pygments.

Also switch to upstream provided github-pages docker clone: The pages label is supposed to be as close to github's rendering as possible. https://github.com/jekyll/docker

This correctly labels the code with the language. However the coloring is still not rendered.

### CODE REVIEW

**Comment by tfoote** [comment]
This removes the need for #80

**Comment by tkruse** [comment]
Along these lines, these news could be interesting:
https://github.com/blog/2100-github-pages-now-faster-and-simpler-with-jekyll-3-0

tl/dr; github will reduce the number of supported formats.

File: articles/080_ros_documentation_system.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ permalink: articles/ros_documentation_system.html
 abstract:
   This article describes the proposed system for doing documentation for ROS 2.
 author: '[William Woodall](https://github.com/wjwwood)'
+published: false
```
Comment (dirk-thomas) [inline articles/080_ros_documentation_system.md:8]: How is this related?

Comment (tfoote) [inline articles/080_ros_documentation_system.md:8]: Jekyll fails to generate without this.

```
  Liquid Exception: cannot load such file -- rouge in articles/080_ros_documentation_system.md
/usr/local/lib/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- rouge (LoadError)
    from /usr/local/lib/site_ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/tags/highlight.rb:102:in `render_rouge'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/tags/highlight.rb:53:in `render'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/block.rb:109:in `block in render_all'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/block.rb:96:in `each'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/block.rb:96:in `render_all'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/block.rb:82:in `render'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/template.rb:128:in `render'
    from /usr/lib/ruby/gems/2.2.0/gems/liquid-2.6.3/lib/liquid/template.rb:138:in `render!'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/convertible.rb:106:in `render_liquid'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/convertible.rb:233:in `do_layout'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/page.rb:122:in `render'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/site.rb:269:in `block in render'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/site.rb:268:in `each'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/site.rb:268:in `render'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/site.rb:46:in `process'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/command.rb:28:in `process_site'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:55:in `build'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:33:in `process'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/lib/jekyll/commands/serve.rb:25:in `block (2 levels) in init_with_program'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
    from /usr/lib/ruby/gems/2.2.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
    from /usr/lib/ruby/gems/2.2.0/gems/jekyll-2.4.0/bin/jekyll:18:in `<main>'
```

File: _config.yml [hunk 2]
```diff
@@ -1,15 +1,17 @@
 name: ROS2 Design
 description: "Distilled design documents related to the ROS 2.0 effort"
 
-baseurl: http://design.ros2.org
+baseurl: ""
```
Comment (dirk-thomas) [inline _config.yml]: Does the value need to be empty or can it be removed?

Comment (tfoote) [inline _config.yml]: It's not necessary, but I think it's better to have as empty since we reference it throughout the code. It's a value which can be tuned so it's better to have the introspection that it's empty.

Comment (dirk-thomas) [inline _config.yml]: If the usage of baseurl is a no-op should we remove the usage?

Comment (tfoote) [inline _config.yml]: It's a standard extension point for jekyll: https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/

We could strip it entirely but I don't think it's worth doing that.

Comment (dirk-thomas) [inline _config.yml]: Well, since it is not a lot of work I think it makes sense: #82.

Comment (tfoote) [inline _config.yml]: I added a commit to remove baseurl to match #82

**Comment by dirk-thomas** [comment]
lgtm
